src/components/header/header-cell.component.ts

```diff
@@ -141,11 +153,31 @@ export class DataTableHeaderCellComponent {
   _sorts: any[];
   selectFn = this.select.emit.bind(this.select);
 
+  private headerTemplateContext: HeaderTemplateContext;
+
+  private initialized: boolean;
+
   @HostListener('contextmenu', ['$event'])
   onContextmenu($event: MouseEvent): void {
     this.columnContextmenu.emit({ event: $event, column: this.column });
   }
 
+  ngOnInit() {
+    this.updateHeaderTemplateContext();
+    this.initialized = true;
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!this.initialized) return;
+    // determine if a property that affects headerTemplateContext has changed
+    for (const prop in headerContextProps) {
+      if (changes.hasOwnProperty(prop)) {
+        this.updateHeaderTemplateContext();
```

Comment:
I thought you needed to create a new object to trigger change detection in the headerTemplate.